app/src/behaviors/behavior_caps_word.c

```diff
@@ -24,33 +24,66 @@ LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
 
 #if DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT)
 
-struct caps_word_continue_item {
-    uint16_t page;
-    uint32_t id;
-    uint8_t implicit_modifiers;
+struct key_list {
+    const struct zmk_key_decoded *keys;
+    size_t len;
 };
 
 struct behavior_caps_word_config {
+    struct key_list continuations;
+    struct key_list shifted;
+    int idle_timeout_ms;
     zmk_mod_flags_t mods;
-    uint8_t index;
-    uint8_t continuations_count;
-    struct caps_word_continue_item continuations[];
+    bool no_default_keys;
 };
 
 struct behavior_caps_word_data {
+    struct k_timer idle_timer;
     bool active;
 };
 
+static const struct device *devs[DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT)];
+static size_t dev_count = 0;
+static int pressed_key_count = 0;
+
+static void restart_caps_word_idle_timer(const struct device *dev) {
+    const struct behavior_caps_word_config *config = dev->config;
+    struct behavior_caps_word_data *data = dev->data;
+
+    if (config->idle_timeout_ms) {
```

Comment:
Any reason not to use a delayable work for this, instead of a timer?

Reply:
Nope. Fixed.